cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -68,36 +70,45 @@ func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) erro
 	}
 
 	// User admin should have full access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "admin", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for admin: %v", err)
+	adminCertConfig := &certutil.Config{
+		CommonName:   AdminKubeConfigClientName,
+		Organization: []string{"system:masters"},
+	}
+	adminKubeConfigFilePath := path.Join(outDir, AdminKubeConfigFileName)
+	if err := createKubeConfigFileForClient(masterEndpoint, adminKubeConfigFilePath, adminCertConfig, caCert, caKey); err != nil {
+		return fmt.Errorf("couldn't create config for %s: %v", AdminKubeConfigClientName, err)
 	}
 
-	// TODO: The kubelet should have limited access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "kubelet", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for kubelet: %v", err)
+	// The kubelet should have limited access to the cluster
+	kubeletCertConfig := &certutil.Config{
+		CommonName:   KubeletKubeConfigClientName,
+		Organization: []string{"system:nodes"},
+	}
+	kubeletKubeConfigFilePath := path.Join(outDir, KubeletKubeConfigFileName)
+	if err := createKubeConfigFileForClient(masterEndpoint, kubeletKubeConfigFilePath, kubeletCertConfig, caCert, caKey); err != nil {
+		return fmt.Errorf("couldn't create config for %s: %v", KubeletKubeConfigClientName, err)
 	}
 
```

Comment:
opened https://github.com/kubernetes/kubernetes/pull/40034, will update kubeadm to reference it once this merged